### ISSUE TITLE
World Corruption After Last Update

### ISSUE THREAD

**Issue by pauloelr** [MinecraftModDevelopmentMods/MMDLib#35]
After last update of *Metals mods
 - EndMetals
 - BaseMetals
 - ModernMetals
 - NetherMetas
 - MMDLib

I received the message on the screenshot and if I proceed after this the world will stuck in the first tick and even if I downgrade the mods the world will be already corrupted forever

<img width="742" alt="Screen Shot 2019-05-20 at 16 53 06" src="https://user-images.githubusercontent.com/1790910/58048082-36cbb300-7b20-11e9-8f1c-6adbc67b5761.png">

Those are the only mods I updated

**Comment by KiriCattus** [comment]
Do you have any console logs? Stick them on [hastebin](https://hastebin.com) see if there's any more info for it.

**Comment by pauloelr** [comment]
This is the crash-report with time most close the the moment of the problem: https://hastebin.com/yevugehodo.rb

But it does not have much information

**Comment by dshadowwolf** [comment]
hastebin paste seems missing - but I think this is because we've had some stuff move around from -beta8 to -rc1 and then to -rc2. 

TL;DR - I made a major mistake heading into -rc1 that was corrected in -rc2...

Basically... In -beta8 BaseMetals was providing everything for the vanilla minecraft integration and extension. When -rc1 was released I'd thought that people might want to use MMDLib with just, say, NetherMetals and have all the vanilla extended stuff still available, so I moved that to MMDLib. This got me in a bit of trouble with the project lead @jriwanek - since the MMDLib should, at her insistence, not provide anything. So for -rc2 I moved things back to BaseMetals... 

The update will cause a backup level.dat to be created - this is because there is remapping of Item and Block "registry names" and possibly ID numbers happening. There should be no actual corruption - in fact, if you were using the same world with -beta8 and previous versions of BaseMetals, then the release of -rc1 is what caused the backup level.dat to be created in the first place and the release of -rc2 is triggering the message as Minecraft has detected this.

You can try going to your instance directory, into the "saves", deleting "level.dat" and then renaming "level.dat_old" to "level.dat" - though I don't recommend it, as you'd lose all progress on the world from the -rc1 to -rc2 updates. A better solution would be to rename the original from "level.dat_old" to something like "level.dat_old_saved" and seeing if there is any actual corruption of the world.

**Comment by pauloelr** [comment]
Yes, I was using the same world with -beta8 and previous versions of BaseMetals, and I tried every option you described here and in all cases it results in World corruption. Also, if i rename the "level.dat_old" to something like "level.dat_old_saved" I end up with this error: https://hastebin.com/idomebarow.rb
